lib/avro-patches/logical_types/schema.rb

```diff
@@ -57,8 +57,8 @@ def self.real_parse(json_obj, names=nil, default_namespace=nil)
   end
 
   # Determine if a ruby datum is an instance of a schema
-  def self.validate(expected_schema, logical_datum, encoded = false)
-    Avro::SchemaValidator.validate!(expected_schema, logical_datum, encoded)
+  def self.validate(expected_schema, logical_datum, recursive = true, encoded = false)
```

Comment:
I added  the`recursive` arg first since the validator code is already on the Avro master branch. The `encoded` option is used by logical types which are still in PR.

I can switch things around to use an `options` hash. The current Avro code on master doesn't have any methods that take an options hash, so we can see what they say when I package this up to contribute it back.